packages/contracts-bedrock/test/setup/CommonTest.sol

```diff
@@ -68,21 +70,27 @@ contract CommonTest is Test, Setup, Events {
             deploy.cfg().setUseInterop(true);
         }
 
+        if (isForkTest) {
+            // Skip any test suite which uses a nonstandard configuration.
+            if (useAltDAOverride || useLegacyContracts || customGasToken != address(0) || useInteropOverride) {
+                vm.skip(true);
+            }
+
+            // Modifying these values on a fork test causes issues.
+            vm.warp(deploy.cfg().l2OutputOracleStartingTimestamp() + 1);
+            vm.roll(deploy.cfg().l2OutputOracleStartingBlockNumber() + 1);
+            vm.fee(1 gwei);
+        }
+
         vm.etch(address(ffi), vm.getDeployedCode("FFIInterface.sol:FFIInterface"));
+        vm.allowCheatcodes(address(ffi));
```

Comment:
Why did we need to add this?

Reply:
Basically because it causes a bunch of tests to not fail. :)

For context, from the [foundry book](https://book.getfoundry.sh/cheatcodes/allow-cheatcodes):

> By default, the test contract, and its deployer are allowed to access cheatcodes. In addition to that, cheat code access is granted if the contract was deployed by an address that already has cheatcode access...

But the FfiInterface contract is not deployed by the test contract, it is etched directly into state [here](https://github.com/ethereum-optimism/optimism/blob/upgrade-against-registry/packages/contracts-bedrock/test/setup/CommonTest.sol#L34).